packages/eslint-plugin/tests/rules/no-unnecessary-condition.test.ts

```diff
@@ -573,6 +573,23 @@ function get<Obj, Key extends keyof Obj>(obj: Obj, key: Key) {
 
 get({ foo: null }, 'foo');
     `,
+    {
+      code: `
+        function foo(val: unknown) {}
+        foo([] === true);
+      `,
+      options: [{ allowAssertTypePredicateParameterConditions: true }],
+    },
+    {
+      code: `
+        function assert(condition: unknown): asserts condition {
+          // do something with condition
+        }
+        const obj = {};
+        assert(obj != null);
+      `,
+      options: [{ allowAssertTypePredicateParameterConditions: true }],
+    },
```

Comment:
Requesting one more `valid` test case: `assert(obj)`. That kind of assertion comes up once in a while.